pkg/kubelet/nodestatus/setters.go

```diff
@@ -781,3 +783,78 @@ func VolumesInUse(syncedFunc func() bool, // typically Kubelet.volumeManager.Rec
 		return nil
 	}
 }
+
+// RuntimeHandlers returns a Setter that sets RuntimeHandlers on the node.
+func SwapEnabledCondition(swapConfig config.MemorySwapConfiguration,
+	isSwapOnFunc func() (bool, error), // typically swap.IsSwapOn
+	nowFunc func() time.Time, // typically Kubelet.clock.Now
+) Setter {
+	return func(_ context.Context, node *v1.Node) error {
+		currentTime := metav1.NewTime(nowFunc())
+		var condition *v1.NodeCondition
+
+		// Check if NodeDiskPressure condition already exists and if it does, just pick it up for update.
+		for i := range node.Status.Conditions {
+			if node.Status.Conditions[i].Type == v1.NodeSwapEnabled {
+				condition = &node.Status.Conditions[i]
+			}
+		}
+
+		newCondition := false
+		// If the NodeSwapEnabled condition doesn't exist, create one
+		if condition == nil {
+			condition = &v1.NodeCondition{
+				Type:   v1.NodeSwapEnabled,
+				Status: v1.ConditionUnknown,
+			}
+			// cannot be appended to node.Status.Conditions here because it gets
+			// copied to the slice. So if we append to the slice here none of the
+			// updates we make below are reflected in the slice.
+			newCondition = true
+		}
+
+		// Update the heartbeat time
+		condition.LastHeartbeatTime = currentTime
+
+		isSwapOn, err := isSwapOnFunc()
+		if err != nil {
+			return fmt.Errorf("error checking if swap is on: %w", err)
+		}
+
+		swapBehavior := swapConfig.SwapBehavior
+		conditionMessage := fmt.Sprintf("is swap provisioned on node: %t. swapBehavior: %s", isSwapOn, swapBehavior)
+
+		setSwapCondition := func(status v1.ConditionStatus) {
+			if condition.Status == status && condition.Message == conditionMessage {
+				return
+			}
+
+			condition.Status = status
+			if status == v1.ConditionTrue {
+				condition.Reason = "SwapEnabled"
+			} else {
+				condition.Reason = "SwapDisabled"
+			}
+			condition.Message = conditionMessage
+			condition.LastTransitionTime = currentTime
+		}
+
+		if isSwapOn && swapBehavior != types.NoSwap {
+			setSwapCondition(v1.ConditionTrue)
+		} else {
+			setSwapCondition(v1.ConditionFalse)
+		}
+
+		if !utilfeature.DefaultFeatureGate.Enabled(features.NodeSwap) {
+			// TODO(iholder101): remove on GA
+			conditionMessage = "NodeSwap feature gate is disabled"
+			setSwapCondition(v1.ConditionFalse)
```

Comment:
I don't feel we should update node conditions for feature gates, in theory features gates disable the code path of the new features, so it should be here https://github.com/kubernetes/kubernetes/pull/125278/files#r1625152968

Reply:
+1
IMO the condition should be present at all only if the FG is enabled

Reply:
Yes! Thank you for this important suggestion @aojea, I fully agree.
Done :+1: 

I believe it also simplified the unit and e2e tests. thank you!